src/TaskRunner/Commands/BuildCommands.php

```diff
@@ -325,66 +323,109 @@ public function buildAssets(ConsoleIO $io, array $options = [
         $finder->directories()
             ->in($options['custom-code-folder'])
             ->name($options['default-theme']);
+        if (!$finder->hasResults()) {
+            $this->say("The theme '{$options['default-theme']}' couldn't be found on the '{$options['custom-code-folder']}' folder.");
+            return 0;
+        }
+        $theme_dir = '';
+        foreach ($finder as $directory) {
+            $theme_dir = $directory->getRealPath();
+        }
+        $files = scandir($theme_dir);
+        $taskRunners = explode(' ', $options['theme-task-runner']);
+        $allowedTaskRunners = [
+            'gulp' => 'gulpfile.js',
+            // Deprecated 'grunt'.
+            'grunt' => 'Gruntfile.js',
+            'ecl-builder' => 'ecl-builder.config.js',
+        ];
+        foreach ($taskRunners as $taskRunner) {
+            if (!in_array($taskRunner, array_keys($allowedTaskRunners))) {
+                $this->say("$taskRunner is not a supported 'theme-task-runner'. The supported plugins are 'ecl-builder' and 'gulp'.");
+                return 0;
+            }
+        }
+        $this->buildAssetsInstall($theme_dir, $allowedTaskRunners, $taskRunners, $files, $options);
+        $this->buildAssetsCompile($taskRunners, $options, $theme_dir);
+        return 1;
```

Comment:
Hi @zarabatana ,
Any reason to return a failure here ?
This will block any further command to be executed, for example, if added to the build-dev post commands, it will fail in the pipeline.